File: php/src/Infrastructure/Persistence/InMemoryGreetingRepository.php
<?php

declare(strict_types=1);

namespace Alxarafe\App\Infrastructure\Persistence;

use Alxarafe\App\Domain\Model\Greeting;
use Alxarafe\App\Domain\Model\GreetingRepositoryInterface;

final class InMemoryGreetingRepository implements GreetingRepositoryInterface
{
    /** @var array<string, Greeting> */
    private array $items = [];

    public function save(Greeting $greeting): void
    {
        $this->items[$greeting->getId()] = $greeting;
    }

    public function findById(string $id): ?Greeting
    {
        return $this->items[$id] ?? null;
    }

    /** @return Greeting[] */
    public function findAll(): array
    {
        return array_values($this->items);
    }
}

File: java/src/main/java/com/alxarafe/app/infrastructure/persistence/InMemoryGreetingRepository.java
package com.alxarafe.app.infrastructure.persistence;

import com.alxarafe.app.domain.model.Greeting;
import com.alxarafe.app.domain.model.GreetingRepository;

import java.util.*;
import java.util.concurrent.ConcurrentHashMap;

public final class InMemoryGreetingRepository implements GreetingRepository {

    private final Map<String, Greeting> items = new ConcurrentHashMap<>();

    @Override
    public void save(Greeting greeting) {
        items.put(greeting.getId(), greeting);
    }

    @Override
    public Optional<Greeting> findById(String id) {
        return Optional.ofNullable(items.get(id));
    }

    @Override
    public List<Greeting> findAll() {
        return List.copyOf(items.values());
    }
}
